### PR TITLE
feat(Deconreconstruction): add presence

### PR DESCRIPTION
**Pull request by SerketCity**
## Description 

Added presence support for the Deconreconstruction website.

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image
image

</details>

### CODE REVIEW

File: websites/D/Deconreconstruction/presence.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+				default:
+					presenceData.details = "Reading Vast Error";
+					presenceData.smallImageKey = "ve";
+					presenceData.state = `Page ${pathArr[2]} of ${vePages}`;
+					presenceData.smallImageText =
+						document.querySelector("#command").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.deconreconstruction.com${pathname}`,
```
Comment (DarkVillager) [inline websites/D/Deconreconstruction/presence.ts:42]:
```suggestion
							url: href,
```

File: websites/D/Deconreconstruction/presence.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+		clientId: "945791824147128341",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname } = document.location,
```
Comment (DarkVillager) [inline websites/D/Deconreconstruction/presence.ts:11]:
```suggestion
		{ href, pathname } = document.location,
```

File: websites/D/Deconreconstruction/presence.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+				case "syzygy":
+					presenceData.details = "Reading Thaumatrope: Syzygy";
+					presenceData.smallImageKey = "thaumatrope";
+					presenceData.state = `Page ${pathArr[3]} of 162`;
+					presenceData.smallImageText =
+						document.querySelector("#command").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.deconreconstruction.com${pathname}`,
```
Comment (DarkVillager) [inline websites/D/Deconreconstruction/presence.ts:60]:
```suggestion
							url: href,
```

File: websites/D/Deconreconstruction/presence.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+				case "haustoria":
+					presenceData.details = "Reading Thaumatrope: Haustoria";
+					presenceData.smallImageKey = "thaumatrope";
+					presenceData.state = `Page ${pathArr[3]} of ${htPages}`;
+					presenceData.smallImageText =
+						document.querySelector("#command").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.deconreconstruction.com${pathname}`,
```
Comment (DarkVillager) [inline websites/D/Deconreconstruction/presence.ts:74]:
```suggestion
							url: href,
```

File: websites/D/Deconreconstruction/presence.ts [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+				case "april-fools-day-2017":
+					presenceData.details = "Reading April Fools 2017";
+					presenceData.smallImageKey = "mspa";
+					presenceData.state = `Page ${pathArr[3]} of 4`;
+					presenceData.smallImageText =
+						document.querySelector("#command").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.deconreconstruction.com${pathname}`,
```
Comment (DarkVillager) [inline websites/D/Deconreconstruction/presence.ts:100]:
```suggestion
							url: href,
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by DarkVillager** [comment]
Also optional you could add the wiki?

File: websites/D/Deconreconstruction/presence.ts [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+(function () {
+	function getPages() {
+		if (document.location.pathname.match(/^\/vasterror\/\d+$/)) {
+			fetch(
+				"https://api.deconreconstruction.com/pages/count?story.name=vast-error&published_at_null=false"
+			)
+				.then(res => res.json())
+				.then(res => {
+					vePages = res;
+				});
```
Comment (EncryptedDev) [inline websites/D/Deconreconstruction/presence.ts:138]: Both of the `.then()`s are not needed as you can merge them together

**Comment by SerketCity** [comment]
> Also optional you could add the wiki?

i intend to do so after this first version is released. theres a couple other subdomains of the site that have wildly different structuring and functionality so i wanted to break it up and release them one at a time

File: websites/D/Deconreconstruction/presence.ts (hunk 6 above)
Comment (SerketCity) [reply websites/D/Deconreconstruction/presence.ts:138]: how would you recommend to merge them? i keep getting type errors 
image
image

Comment (DarkVillager) [reply websites/D/Deconreconstruction/presence.ts:138]: await?

Comment (RisingSunLight42) [reply websites/D/Deconreconstruction/presence.ts:138]: Yeah, await

Comment (SerketCity) [reply websites/D/Deconreconstruction/presence.ts:138]: i figured as much, but i dont know where to put the await without it telling me this
image
(even when i make the function async)

Comment (RisingSunLight42) [reply websites/D/Deconreconstruction/presence.ts:138]: Transform your function into an async one.

Comment (SerketCity) [reply websites/D/Deconreconstruction/presence.ts:138]: it still marks the await as out of place when making either/both of the functions that its within as async

Comment (RisingSunLight42) [reply websites/D/Deconreconstruction/presence.ts:138]: Works for me, try to start an auto-fix, it should be proposed

**Review by EncryptedDev — APPROVED** [review]

File: websites/D/Deconreconstruction/presence.ts [hunk 7]
```diff
@@ -0,0 +1,149 @@
+const presence = new Presence({
+		clientId: "945791824147128341",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
```
Comment (Bas950) [inline websites/D/Deconreconstruction/presence.ts:8]: please use an imgur link for future maintainability

File: websites/D/Deconreconstruction/presence.ts [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+					presenceData.details = "Viewing the credits";
+					break;
+
+				case "log":
+					presenceData.details = "Viewing the Vast Error log";
+					break;
+
+				default:
+					presenceData.details = "Reading Vast Error";
+					presenceData.smallImageKey = "ve";
```
Comment (Bas950) [inline websites/D/Deconreconstruction/presence.ts:35]: same here

File: websites/D/Deconreconstruction/presence.ts [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+					];
+					break;
+			}
+			break;
+
+		case "thaumatrope":
+			switch (pathArr[2]) {
+				case "syzygy":
+					presenceData.details = "Reading Thaumatrope: Syzygy";
+					presenceData.smallImageKey = "thaumatrope";
```
Comment (Bas950) [inline websites/D/Deconreconstruction/presence.ts:53]: and here

File: websites/D/Deconreconstruction/presence.ts [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+						{
+							label: "Read Along",
+							url: href,
+						},
+					];
+					break;
+
+				case "haustoria":
+					presenceData.details = "Reading Thaumatrope: Haustoria";
+					presenceData.smallImageKey = "thaumatrope";
```
Comment (Bas950) [inline websites/D/Deconreconstruction/presence.ts:67]: here.

File: websites/D/Deconreconstruction/presence.ts [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+					presenceData.details = "Viewing an unsupported page";
+					break;
+			}
+			break;
+
+		case "bonus":
+			switch (pathArr[2]) {
+				case "april-fools-day-2017":
+					presenceData.details = "Reading April Fools 2017";
+					presenceData.smallImageKey = "mspa";
```
Comment (Bas950) [inline websites/D/Deconreconstruction/presence.ts:93]: one more

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**